### ISSUE TITLE
Empty repository shows panels

### ISSUE THREAD

**Issue by RussKie** [gitextensions/gitextensions#5249]
**Do you want to request a *feature* or report a *bug*?**
bug / visual inconsistency

**What is the current behavior?**
Whenever an empty repository is opened, all information panels (left/bottom/right) must be hidden.
The toggle buttons probably must be disabled as well.

![image](https://user-images.githubusercontent.com/4403806/43429791-cc40afd0-94a8-11e8-8ba5-7485d7eac489.png)

**What is the expected behavior?**

No panels are visible/enabled until a repository with commits is opened.
The state of panels must be restored to what it was before an empty repository is opened (e.g. the state must not be persisted into `AppSettings`).

**Environment you encounter the issue:**

- GitExtensions version:
- GIT version: master @ 278c22c43ae33fd41f8fa4939cbe0dd26eec0e04

**Comment by drewnoakes** [comment]
💭 The console and the remote tree could be useful in an empty repo.

**Comment by RussKie** [comment]
how so?

**Comment by vbjay** [comment]
Fetching from remotes.  Showing what remotes and their branches are.

**Comment by drewnoakes** [comment]
Like @vbjay says, the repo might be empty because you haven't fetched its contents from anywhere. You might do this from the console, or via a remote.

It's not a huge deal. We could just hide everything, or hide the info/diff/tree tabs.

Was playing around with this and found #5251.

Longer term I wonder if we might hoist the terminal to a new top-level tab group that contains both browse and commit tabs, potentially for multiple repos in one window.

**Comment by RussKie** [comment]
> Fetching from remotes.  Showing what remotes and their branches are.

Thanks, good point.
Is there a value in showing the bottom panel?

**Comment by vbjay** [comment]
Yes. Just because they aren't checked out doesn't mean I shouldn't be able
to compare commits, and run cmds and see got info and ....

**Comment by drewnoakes** [comment]
> Just because they aren't checked out doesn't mean I shouldn't be able to compare commits, and run cmds and see got info and ....

In the scenario we're talking about the repo has zero commits. There are no revisions to view. I don't think the info/diff/tree tabs make sense in such a case.

I don't think the Branches node makes sense either, until at least one commit exists.

**Comment by vbjay** [comment]
Oh right.  Sorry bare vs empty.

**Comment by gerhardol** [comment]
> Longer term I wonder if we might hoist the terminal to a new top-level tab group that contains both browse and commit tabs, potentially for multiple repos in one window.

The good thing about the Console tab is that it follows the repo. That is the reason I use it, for normal work I have other consoles. So the Console should not be shared between repos.

I could see a reason to put it in the right panel - then I can see the revision tree and diff tab.

**Comment by RussKie** [comment]
Do we all agree that all panels should be hidden until there is at least one commit?

**Comment by drewnoakes** [comment]
Your call RussKie. If we go the simple route now, let's at least track the tougher route later. Honestly, users spend a miniscule fraction of their time in GE seeing an empty repo. Right now we can end up showing a commit from the previously opened non-empty repo in the lower panel, which looks broken. At the very least, we should fix that.

**Comment by gerhardol** [comment]
> Do we all agree that all panels should be hidden until there is at least one commit?

I find this so low prio that I rather do no changes than risk breaking something else
